presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/PrestoExceptionClassifier.java

```diff
@@ -104,20 +106,23 @@
     private final Map<Integer, ErrorCodeSupplier> errorByCode;
     private final Set<ErrorCodeSupplier> retryableErrors;
 
-    public PrestoExceptionClassifier(
-            Set<ErrorCodeSupplier> additionalErrors,
-            Set<ErrorCodeSupplier> additionalRetryableErrors)
+    private PrestoExceptionClassifier(Set<ErrorCodeSupplier> recognizedErrors, Set<ErrorCodeSupplier> retryableErrors)
     {
-        this.errorByCode = ImmutableSet.<ErrorCodeSupplier>builder()
-                .addAll(DEFAULT_ERRORS)
-                .addAll(additionalErrors)
-                .build()
-                .stream()
+        this.errorByCode = recognizedErrors.stream()
                 .collect(toImmutableMap(errorCode -> errorCode.toErrorCode().getCode(), identity()));
-        this.retryableErrors = ImmutableSet.<ErrorCodeSupplier>builder()
-                .addAll(DEFAULT_RETRYABLE_ERRORS)
-                .addAll(additionalRetryableErrors)
-                .build();
+        this.retryableErrors = ImmutableSet.copyOf(retryableErrors);
+    }
+
+    public static PrestoExceptionClassifier createDefault()
+    {
+        return create(DEFAULT_ERRORS, DEFAULT_RETRYABLE_ERRORS);
+    }
+
+    public static PrestoExceptionClassifier create(Set<ErrorCodeSupplier> recognizedErrors, Set<ErrorCodeSupplier> retryableErrors)
```

Comment:
Why have this in place of the default constructor. The arguments are the same. We can still have a `PrestoExceptionClassifier.default()` method for the default methods.

Reply:
We can do either, but I think usually when we use factory constructor for a class, we make the constructor private and require construction only through the factory constructors, instead of allowing a mix of regular constructors and factory constructors.